mlir/include/mlir/Dialect/GPU/IR/GPUOps.td

```diff
@@ -550,12 +601,23 @@ def GPU_LaunchFuncOp :GPU_Op<"launch_func", [
     /// The name of the kernel.
     StringAttr getKernelName();
 
+    /// Returns true if cluster size is specified.
+    bool hasClusterSize() {
+      if (getClusterSizeX() && getClusterSizeY() && getClusterSizeZ())
```

Comment:
Is it valid to set `X`, but not `Y` and `Z`?

I'm thinking we could check if one is set and assert that the others are set too.

Reply:
All dimensions must be explicitly defined just like block or grid dimensions.
(We could implicitly the non-used dimensions set 1)